core/src/main/scala/weaponregex/parser/Parser.scala

```diff
@@ -8,26 +8,46 @@ import weaponregex.extension.StringExtension.StringIndexExtension
 
 import scala.util.{Failure, Success, Try}
 
-/** Companion object for [[weaponregex.parser.Parser]] class that instantiates [[weaponregex.parser.Parser]] instances
+/** Companion object for [[weaponregex.parser.Parser]] class that instantiates flavor-specific parsers instances
   */
 object Parser {
 
-  /** Regex special characters
-    */
-  final private val specialChars: String = """()[{\.^$|?*+"""
-
   /** Apply the parser to parse the given pattern
     * @param pattern The regex pattern to be parsed
     * @return A `Success` of parsed [[weaponregex.model.regextree.RegexTree]] if can be parsed, a `Failure` otherwise
     */
-  def apply(pattern: String): Try[RegexTree] = new Parser(pattern).parse
+  def apply(pattern: String, flavor: ParserFlavor = ParserFlavorJVM): Try[RegexTree] = flavor match {
+    case ParserFlavorJVM => new ParserJVM(pattern).parse
+    case ParserFlavorJS  => new ParserJS(pattern).parse
+    case _               => Failure(new RuntimeException("[Error] Parser: Unsupported regex flavor"))
```

Comment:
Since ParserFlavor is a sealed trait, isn't it impossible to get into this case? One of the first two will always match

Reply:
It probably is, but we added that anyway, as matches need to be exhaustive

Reply:
Does the compiler give a warning? For sealed traits it shouldn't be needed as the compiler can figure out what all the possible options are (and if you've checked for them all)